angular-client/src/app/app-nav-bar/app-nav-bar.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, HostListener, inject, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { MessageService } from 'primeng/api';
+import { interval, map, Observable, startWith } from 'rxjs';
+import { startNewRun } from 'src/api/run.api';
+import APIService from 'src/services/api.service';
+import SidebarService from 'src/services/sidebar.service';
+import Storage from 'src/services/storage.service';
+
+interface NavItem {
+  label: string;
+  route: string;
+  icon: string;
+}
+
+@Component({
+  selector: 'app-nav-bar',
+  templateUrl: './app-nav-bar.component.html',
+  styleUrls: ['./app-nav-bar.component.css']
+})
+export class AppNavBarComponent implements OnInit {
+  private storage = inject(Storage);
+  private serverService = inject(APIService);
+  private messageService = inject(MessageService);
+  private router = inject(Router);
+  location: string = 'Boston, Massachusetts';
```

Comment:
this shouldnt be hardcoded, either add TODO to get this from server or do it here